.github/workflows/vue-todo.yml

```diff
@@ -0,0 +1,78 @@
+name: Example App
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - features/*
+  pull_request:
+    branches:
+      - dev
+    paths-ignore:
+      - 'webview-src/**'
+      - 'webview-dist/**'
+      - 'examples/todos-app/**'
+
+jobs:
+  test-tauri:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [macos-latest, ubuntu-latest, windows-latest]
+
+    runs-on: ${{ matrix.platform }}
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Setup node
+      uses: actions/setup-node@v1
+      with:
+        node-version: 16
+
+    - name: Install Rust stable
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+
+    - name: Install webkit2gtk (ubuntu only)
+      if: matrix.platform == 'ubuntu-latest'
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y webkit2gtk-4.0
+
+    - name: Cache cargo registry
```

Comment:
i know we do this in test.yml already, but is there an actual advantage to splitting this in 3 caches instead of one?
We could basically replace all this with a single one-liner `- uses: Swatinem/rust-cache@v1` (or 5 lines if we keep using actions/cache directly 🤷 )

Reply:
I was assuming that certain deps were platform specific so you'd have to isolate the caches but maybe this isn't the case?

Reply:
In fact I suspect the dep on the C++ Sharp image conversion library is an example of this, no?

Reply:
You're absolutely correct. different OSes should have seperate caches. Breaking issues aside, this would make them much less effective. They would either increase in size and therefore increase the compressing/uncompressing times or would be kinda useless because of race conditions (for example it would be possible that only the macOS build files would end up in there).

But this is also not what i was trying to say. What i meant is whether there would be a disadvantage in merging the 3 steps for setting up the caches for .cargo/registry, .cargo/git and target/ into one single workflow step.
For example [here](https://github.com/FabianLars/mw-toolbox/blob/main/.github/workflows/ci.yml#L75) i'm doing effectively the same, in a single line which is much cleaner imo.

I also never saw that split (into 3) in the wild either, other than in tauri repos, so in the end i'm just curious myself if there are actualy pros/cons.